app/src/composables/use-flows.ts

```diff
@@ -0,0 +1,284 @@
+
+		if (!currentFlow.value.options?.requireConfirmation) return null;
+
+		return {
+			description: currentFlow.value.options.confirmationDescription,
+			fields: (currentFlow.value.options.fields ?? []).map((field: Record<string, any>) => ({
+				...field,
+				name: !field.name && field.field ? formatTitle(field.field) : field.name,
+			})),
+		};
+	});
+
+	const displayCustomConfirmDialog = computed(
+		() =>
+			!!currentFlowId.value &&
+			!!confirmDialogDetails.value &&
+			!!currentFlowConfirmations.value?.isCustomDialogConfirmationRequired &&
+			!currentFlowConfirmations.value?.isCustomDialogConfirmed &&
+			!displayUnsavedChangesDialog.value,
+	);
+
+	const displayUnsavedChangesDialog = computed(
+		() =>
+			!!currentFlowId.value &&
+			!!currentFlowConfirmations.value?.isUnsavedChangesConfirmationRequired &&
+			!currentFlowConfirmations.value?.isUnsavedChangesConfirmed,
+	);
+
+	const isConfirmButtonDisabled = computed(() => {
+		if (!currentFlowId.value) return true;
+
+		for (const field of confirmDialogDetails.value?.fields || []) {
+			if (
+				field.meta?.required &&
+				(!confirmValues.value ||
+					confirmValues.value[field.field] === null ||
+					confirmValues.value[field.field] === undefined)
+			) {
+				return true;
+			}
+		}
+
+		return false;
+	});
+
+	const manualFlows = computed(() => {
+		const manualFlows = flowsStore
+			.getManualFlowsForCollection(collection.value)
+			.filter(
+				(flow) =>
+					!flow.options?.location || flow.options?.location === 'both' || flow.options?.location === location.value,
+			)
+			.map((flow) => ({
+				...flow,
+				options: flow.options ? translate(flow.options) : null,
+				tooltip: getFlowTooltip(flow),
+				isFlowDisabled: checkFlowDisabled(flow),
+				isFlowRunning: checkFlowRunning(flow),
+			}));
+
+		function getFlowTooltip(manualFlow: FlowRaw) {
+			if (location.value === 'item') return t('run_flow_on_current');
+
+			if (manualFlow.options?.requireSelection === false && selection.value.length === 0) {
+				return t('run_flow_on_current_collection');
+			}
+
+			return t('run_flow_on_selected', selection.value.length || 0);
+		}
+
+		function checkFlowDisabled(manualFlow: FlowRaw) {
+			if (location.value === 'item' || manualFlow.options?.requireSelection === false) return false;
+			return !primaryKey?.value && selection.value.length === 0;
+		}
+
+		function checkFlowRunning(manualFlow: FlowRaw) {
+			if (!manualFlow) return false;
+			return runningFlows.value.includes(manualFlow.id);
+		}
+
+		return manualFlows;
+	});
+
+	function resetConfirm() {
+		currentFlowId.value = null;
+		confirmValues.value = null;
+		confirmedUnsavedChanges.value = false;
+		confirmedCustomDialog.value = false;
+	}
+
+	function confirmUnsavedChanges(flowId: string) {
+		confirmedUnsavedChanges.value = true;
+
+		if (!confirmDialogDetails.value) {
+			runManualFlow(flowId);
+		}
+	}
+
+	function confirmCustomDialog(flowId: string) {
+		confirmedCustomDialog.value = true;
```

Comment:
Let’s prevent the function from being executed when disabled. This will prevent `@apply="confirmCustomDialog(currentFlowId.value!)"` from being executed via shortcut (meta+enter) when `isConfirmButtonDisabled.value` is true.

```suggestion
		if (isConfirmButtonDisabled.value) return;

		confirmedCustomDialog.value = true;
```